pinot-controller/src/main/java/org/apache/pinot/controller/helix/core/PinotHelixResourceManager.java

```diff
@@ -2154,13 +2180,39 @@ public boolean addNewReloadAllSegmentsJob(String tableNameWithType, String jobId
     jobMetadata.put(CommonConstants.ControllerJob.JOB_TYPE, ControllerJobType.RELOAD_ALL_SEGMENTS.toString());
     jobMetadata.put(CommonConstants.ControllerJob.SUBMISSION_TIME_MS, Long.toString(System.currentTimeMillis()));
     jobMetadata.put(CommonConstants.ControllerJob.MESSAGE_COUNT, Integer.toString(numberOfMessagesSent));
-    return addControllerJobToZK(jobId, jobMetadata);
+    return addControllerJobToZK(jobId, jobMetadata, ZKMetadataProvider.constructPropertyStorePathForControllerJob(),
+            _propertyStore);
   }
 
-  private boolean addControllerJobToZK(String jobId, Map<String, String> jobMetadata) {
-    String jobResourcePath = ZKMetadataProvider.constructPropertyStorePathForControllerJob();
+  public static boolean addRebalanceResultToZk(String tableNameWithType,
```

Comment:
I don't think this method belongs to this class. Suggest moving it to `ZkBasedTableRebalanceObserver` and change it to non-static method. All the arguments should already be member variable